### PR TITLE
[M] Fixed an exception in CertificateUtil when using large certificates

### PR DESCRIPTION
**Pull request by Ceiu**
- Fixed a bug in the CertificateUtil spec test utility which would fail to extract and decode the entitlement body of an entitlement certificate (SCA or otherwise) if the entitlement body was too large and exceeded the hard-coded ~48k limit

### CODE REVIEW

File: spec-tests/src/main/java/org/candlepin/spec/bootstrap/data/util/CertificateUtil.java [hunk 1]
```diff
@@ -141,33 +150,25 @@ public static JsonNode decodeAndUncompressCertificate(String certificate, Object
...
         certificate = certificate.replace("\"", "")
             .replace("\\n", Character.toString((char) 10));
 
         // Retrieve the compressed data body
-        certificate = certificate.split("-----BEGIN ENTITLEMENT DATA-----")[1];
-        certificate = certificate.split("-----END ENTITLEMENT DATA-----")[0];
-        byte[] compressedBody = fromBase64(certificate.getBytes());
+        String[] chunks = REGEX_ENTITLEMENT_DATA.split(certificate, 3);
+        if (chunks.length < 3) {
+            return null;
```
Comment (joshmalbrecht) [inline spec-tests/src/main/java/org/candlepin/spec/bootstrap/data/util/CertificateUtil.java:160]: Should we throw an exception here instead of returning null?

**Review by joshmalbrecht — COMMENTED** [review]
Looks good! I just had a small question

File: spec-tests/src/main/java/org/candlepin/spec/bootstrap/data/util/CertificateUtil.java (hunk 1 above)
Comment (Ceiu) [reply spec-tests/src/main/java/org/candlepin/spec/bootstrap/data/util/CertificateUtil.java:160]: I thought about it, but a certificate not having an entitlement payload is itself not an error case so it felt weird to fail in that way there. I thought it'd be more useful for any affected tests to be able to catch the null and throw an assertion error than propagating a runtime exception.

Comment (joshmalbrecht) [reply spec-tests/src/main/java/org/candlepin/spec/bootstrap/data/util/CertificateUtil.java:160]: Ah yeah, you are right! +1

**Review by joshmalbrecht — APPROVED** [review]
